windows_service/datadog_checks/windows_service/windows_service.py

```diff
@@ -3,14 +3,134 @@
 # Licensed under a 3-clause BSD style license (see LICENSE)
 import re
 
+import pywintypes
 import win32service
-from six import iteritems
 
 from datadog_checks.base import AgentCheck
 
 SERVICE_PATTERN_FLAGS = re.IGNORECASE
 
 
+class ServiceFilter(object):
+    def __init__(self, name=None, startup_type=None):
+        self.name = name
+        self.startup_type = startup_type
+
+        self._init_patterns()
+
+    def _init_patterns(self):
+        if self.name is not None:
+            self._name_re = re.compile(self.name, SERVICE_PATTERN_FLAGS)
+        if self.startup_type is not None:
+            self._startup_type_re = re.compile(self.startup_type, SERVICE_PATTERN_FLAGS)
```

Comment:
Should we trap and report compilation errors

Reply:
The base check already traps it, but I think we can improve the readability
![image](https://user-images.githubusercontent.com/2266830/190519520-c25cbc39-3a78-49db-8fa3-c82cc926c1e1.png)

Reply:
new error string includes the faulting pattern
![image](https://user-images.githubusercontent.com/2266830/190529836-67fe13a4-5501-4a5c-80cc-dedaa6a5f92e.png)

Reply:
👍